examples/blame_example.ipynb

```diff
@@ -67,212 +64,233 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 4,
-   "metadata": {},
+   "execution_count": null,
+   "metadata": {
+    "id": "kT3m0-zoNdvW"
+   },
    "outputs": [],
    "source": [
     "location_info_response = iai.location_info(location=location)\n",
     "rendered_static_map = location_info_response.birdview_image.decode()\n",
-    "scene_plotter = iai.utils.ScenePlotter(rendered_static_map, \n",
-    "                                       location_info_response.map_fov, \n",
-    "                                       (location_info_response.map_center.x, location_info_response.map_center.y), \n",
+    "scene_plotter = iai.utils.ScenePlotter(rendered_static_map,\n",
+    "                                       location_info_response.map_fov,\n",
+    "                                       (location_info_response.map_center.x, location_info_response.map_center.y),\n",
     "                                       location_info_response.static_actors)"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
-   "metadata": {},
-   "outputs": [
-    {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "[((0, 3), (89, 96), (0,)), ((2, 17), (135, 150), (2,)), ((3, 5), (150, 162), (3,)), ((4, 11), (1, 15), (11,)), ((4, 21), (61, 61), (21,)), ((11, 21), (16, 29), (21,)), ((12, 17), (193, 196), (12,)), ((17, 22), (120, 121), (22,))]\n"
-     ]
-    }
-   ],
+   "execution_count": null,
+   "metadata": {
+    "id": "L0ufxMO8NdvX",
+    "scrolled": true
+   },
+   "outputs": [],
    "source": [
-    "collisions = None\n",
+    "collisions = {}\n",
+    "\n",
+    "@dataclass\n",
+    "class LogCollision:\n",
+    "    collision_agents: Tuple[int, int]\n",
+    "    start_time: int \n",
+    "    end_time: int\n",
+    "\n",
+    "def compute_pairwise_collisions(agent_states,agent_attributes):\n",
```

Comment:
Why do you prefer writing a new function here instead of using the existing function from the `driving_model` repo? It feels like reinventing the wheel, and make the code base hard to manage.

Reply:
That particular function appears to made for performing collision detection in a specific, inapplicable scenario that requires pre-processed collision information as input that is not available in this demo notebook. The function I have provided is more general and takes our standard, common IAI data types as input (e.g. AgentState and AgentAttributes) making it more accessible for clients and other users. Furthermore it is a bit lighter-weight and more transparent how the collision is detected. 

I do not think we will be providing this as a tool for general use so this function would only exist in this notebook anyways and we would furthermore not be importing from driving_models since this notebook needs to work for clients who won't have access to that repository.